reports/report_builder.py

```diff
@@ -4280,8 +4280,10 @@ def ws_sr01(self, ws):
                     # Newspaper journos don't have roles
                     rows = rows.filter(role__in=role_ids)
 
+                rows = self.apply_weights(rows, model.sheet_db_table(), media_type)
+
                 for row in rows:
-                    region_id = [r[0] for r in all_regions if r[1] == row[region]][0]
+                    region_id = [r[0] for r in all_regions if r[1] == row["region"]][0]
```

Comment:
This bit still seems to confuse me. why are we using `row["region"]` instead of `row[region]` given that this is what we define in `.values('sex', region)`?

Reply:
Anybody who tries to understand code at midnight on a Friday night will be confused.

Haven't dug deep but it has do to with how [`dictfetchall`](https://docs.djangoproject.com/en/2.2/topics/db/sql/#executing-custom-sql-directly) works: It fetches data as well as column names. The table column name is `"region"` and not `region` (`model.sheet_name() + '__country_region__region'`).

`apply_weight` uses `dictfetchall` so everything after `apply_weight` is called has to use table column names.